WalletWasabi/Affiliation/Models/CoinjoinRequest/Payload.cs

```diff
@@ -0,0 +1,25 @@
+using Newtonsoft.Json;
+using WalletWasabi.Affiliation.Serialization;
+using System.Text;
+
+namespace WalletWasabi.Affiliation.Models.CoinjoinRequest;
+
+public record Payload
+{
+	public Payload(Header header, Body body)
+	{
+		Header = header;
+		Body = body;
+	}
+
+	[JsonProperty(PropertyName = "header")]
+	public Header Header { get; }
+
+	[JsonProperty(PropertyName = "body")]
+	public Body Body { get; }
+
+	public byte[] GetCanonicalSerialization()
+	{
+		return Encoding.ASCII.GetBytes(JsonConvert.SerializeObject(this, CanonicalJsonSerializationOptions.Settings));
+	}
```

Comment:
Do you mean as a replacement for `OrderedContractResolver`? I can't see any advantage in using `JsonPropertyOrder` instead of `OrderedContractResolver`. Is there any?

Reply:
Ok, I see your point.

> PS: I'm fine with OrderedContractResolver but I would add more tests for this (I wrote a comment somewhere with more tests).

I think [this test](https://github.com/trezor/WalletWasabi/blob/onvej-sl/affiliate-server/WalletWasabi.Tests/UnitTests/Affiliation/CanonicalSerializationTests.cs#L29-L36) is sufficient since it tests that the only structure that I need to be canonically serialized is serialized correctly which is equivalent to specifying `[JsonProperty(Order = number)]` for each property.

Reply:
> Are you against having this test https://github.com/zkSNACKs/WalletWasabi/pull/9769#discussion_r1059682429 too?

In my opinion, the test doesn't bring anything extra that is relevant compared to the test I mentioned.

> I mean it shows how it behaves wrt JsonProperty attribute which is not obvious imho.

This is already tested in the test I mentioned.

Reply:
> But do you honesly expect anybody to read this unformatted blob of characters

No i do not. The blob was generated by the affiliate server and the test checks that both the coordinator and the affiliate-server serialize the request in the same way. This is the only thing I consider to be relevant.

If I were to accept that your test does bring something extra that is relevant compared to my test, I would have to ask why it doesn't test for example sorting of
  * properties encoded in a unicode encoding,
  * nested objects,
  * properties of data types other than `string`, `int`, `decimal`, `bool` and `byte[]`,
  * properties with special characters other than an underscore and at sign (for example a space).

> The goal of tests is to persuade readers that some functionality is correct. 

This is a nice definition but it says nothing unless you specify what the functionality is.

Reply:
Are you OK with https://github.com/zkSNACKs/WalletWasabi/pull/9769/commits/28668f619142e026eaee035d252ac8e363eb19d0?

Reply:
Does it mean this issue is resolved?